src/app/app.config.ts

```diff
@@ -1,15 +1,19 @@
-import { ApplicationConfig, provideExperimentalZonelessChangeDetection } from '@angular/core';
-import { provideRouter, withComponentInputBinding } from '@angular/router';
+import {ApplicationConfig, provideExperimentalZonelessChangeDetection} from '@angular/core';
+import {provideRouter, withComponentInputBinding} from '@angular/router';
 
-import { routes } from './app.routes';
-import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
-import { provideHttpClient, withFetch } from '@angular/common/http';
+import {routes} from './app.routes';
+import {provideAnimationsAsync} from '@angular/platform-browser/animations/async';
+import {provideHttpClient, withFetch, withInterceptors} from '@angular/common/http';
+import {authInterceptor} from "./auth/auth-http-interceptor";
 
 export const appConfig: ApplicationConfig = {
   providers: [
     provideExperimentalZonelessChangeDetection(),
     provideRouter(routes, withComponentInputBinding()),
     provideAnimationsAsync(),
-    provideHttpClient(withFetch())
+    provideHttpClient(
+      withFetch(),
+      withInterceptors([authInterceptor])
```

Comment:
This is adding the auth interceptor to the default `HttpClient`. Currently, this is only injected into the API services, but in the future if we need an `HttpClient` for other purposes, I'll make the default unauthenticated, and setup another just for use in the APIs.

Reply:
Cool